packages/web-app-files/src/components/SideBar/Shares/Collaborators/InviteCollaborator/ExpirationDatepicker.vue

```diff
@@ -1,58 +1,44 @@
 <template>
   <div class="oc-flex oc-flex-middle oc-flex-nowrap">
-    <oc-datepicker
-      v-model="dateCurrent"
-      :min-date="dateMin"
-      :max-date="dateMax"
-      :locale="language.current"
-      :is-required="enforced"
-      class="files-recipient-expiration-datepicker oc-width-1-1"
-      data-testid="recipient-datepicker"
-    >
-      <template #default="{ togglePopover }">
-        <oc-button
-          class="files-collaborators-expiration-button oc-p-s action-menu-item"
-          data-testid="recipient-datepicker-btn"
-          appearance="raw"
-          gap-size="none"
-          :aria-label="
-            dateCurrent ? $gettext('Edit expiration date') : $gettext('Set expiration date')
-          "
-          @click="togglePopover"
-        >
-          <oc-icon name="calendar-event" fill-type="line" size="medium" variation="passive" />
-          <span
-            v-if="!dateCurrent"
-            key="no-expiration-date-label"
-            v-text="$gettext('Set expiration date')"
-          />
-          <span
-            v-else
-            key="set-expiration-date-label"
-            v-text="$gettext('Expires %{expires}', { expires: dateExpire })"
-          />
-        </oc-button>
-      </template>
-    </oc-datepicker>
     <oc-button
-      v-if="!enforced && dateCurrent"
-      class="recipient-edit-expiration-btn-remove"
+      data-testid="recipient-datepicker-btn"
+      class="files-collaborators-expiration-button oc-p-s action-menu-item"
       appearance="raw"
-      :aria-label="$gettext('Remove expiration date')"
-      @click="dateCurrent = null"
+      gap-size="none"
+      :aria-label="dateCurrent ? $gettext('Edit expiration date') : $gettext('Set expiration date')"
+      @click="showDatePickerModal"
     >
-      <oc-icon name="close" />
+      <oc-icon name="calendar-event" fill-type="line" size="medium" variation="passive" />
+      <span
+        v-if="!dateCurrent"
+        key="no-expiration-date-label"
+        v-text="$gettext('Set expiration date')"
+      />
+      <span
+        v-else
+        key="set-expiration-date-label"
+        v-text="$gettext('Expires %{expires}', { expires: dateExpire })"
```

Comment:
Same here, `Edit expiration` might be the better label, although I like to see the expiration duration here :-(

Reply:
Follow up